lib/apipie_bindings/authenticators/negotiate.rb

```diff
@@ -0,0 +1,47 @@
+require 'apipie_bindings/authenticators/base'
+
+module ApipieBindings
+  module Authenticators
+    # Negotiate authenticator
+    # Implements gssapi negotiation with preexisting kerberos ticket
+    # Requires a authentication url, the authentication request will be against.
+    # This url needs to support auth negotiation and after successful auth it should return 'set-cookie' header with session.
+    # Thus this session will be initiated in the auth request and the original request will be made with this cookie.
+    # Next requests should be already skip the negotiation, please implement Session support in your client, for not using the negotiation on every request.
+    class Negotiate < Base
+
+      # Creates new
+      # @param [String] url to make authentication request to.
+      # @param [Hash] auth_request_options passed to RestClient::Request - especially for SSL options
+      #                                    see https://github.com/rest-client/rest-client/blob/master/lib/restclient/request.rb.
```

Comment:
the spaces here are on purpose?

Reply:
Well they are, so it goes after the param name, but I've got no idea what would be the best practice here.